### PR TITLE
[Enhancement] Modify Statistics Resource Usage Policy

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

According to user-configured parallelism. If the number of queries is large, try to piece together multiple queries for UNION (the purpose is to reduce the number of inserts). If there are few queries, increase the dop appropriately to speed up the parallelism of a single query.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 13 (46.15%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/FullStatisticsCollectJob.java|3|10|30.00%|[42, 43, 46, 47, 48, 50, 53]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|2|2|100.00%|[]|

**Review by Seaven — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/statistic/FullStatisticsCollectJob.java [hunk 1]
```diff
@@ -38,11 +38,19 @@ public List<Long> getPartitionIdList() {
...
     @Override
     public void collect(ConnectContext context, AnalyzeStatus analyzeStatus) throws Exception {
-        List<String> collectSQLList = buildCollectSQLList();
         long finishedSQLNum = 0;
+        int parallelism = Math.max(1, context.getSessionVariable().getStatisticCollectParallelism());
+        List<List<String>> collectSQLList = buildCollectSQLList(parallelism);
         long totalCollectSQL = collectSQLList.size();
 
-        for (String sql : collectSQLList) {
+        for (List<String> sqlUnion : collectSQLList) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/statistic/FullStatisticsCollectJob.java:46]: please comment why do we need to change the pipeline dop

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 13 (46.15%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/FullStatisticsCollectJob.java|3|10|30.00%|[42, 43, 53, 54, 55, 57, 60]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|2|2|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
https://github.com/Mergifyio backport branch-2.4

**Pull request merged**